.github/workflows/main.yml

```diff
@@ -86,24 +263,80 @@ jobs:
           MYSQL_PASSWORD: ibis
         options: --health-cmd="mysqladmin ping" --health-interval 10s --health-timeout 5s --health-retries 3
     steps:
-    - name: checkout
-      uses: actions/checkout@v1
-
-    - name: set up environment
-      run: ./ci/setup_env.sh "${{ matrix.python_version }}" "$BACKENDS"
-
-    - name: run tests
-      run: PYTEST_BACKENDS=$BACKENDS ./ci/run_tests.sh -m "not udf"
-
-    - name: pytest errors
-      run: ./ci/pytest_errors.sh
-      if: failure()
-
-  Tests_impala_clickhouse:
-    name: Tests Impala / Clickhouse
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: download env file
+        uses: actions/download-artifact@v2
+        with:
+          name: conda-env-file
+          path: tmp
+
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          mamba-version: "*"
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
+          channel-priority: strict
+          activate-environment: ibis
+          python-version: ${{ matrix.python-version }}
+          environment-file: tmp/environment.yaml
+          condarc-file: ci/condarc
+
+      - name: install pyspark
+        run: |
+          conda install -n ibis \
```

Comment:
Preferably we should use https://github.com/marketplace/actions/setup-apache-spark

Reply:
That actually isn't necessary, just a modern java runtime is needed. The pyspark jobs are now passing